tests/test_time.py

```diff
@@ -64,6 +64,12 @@ def test_set_max_lag_incorrect_val(self):
         with pytest.raises(ValueError):
             cal.set_max_lag(-1)
 
+    def test_visualize(self, dummy_calendar):
+        dummy_calendar.visualize()
+
+    def test_visualize_with_text(self, dummy_calendar):
+        dummy_calendar.visualize(add_freq=True)
```

Comment:
Writing image comparison tests for the plots is going to be messy and it might be too early now. These tests are there just to ensure there is no immediate error due to a change in the format of the intervals, or interface of the Calendar class.